pkg/monitors/uptimerobot/uptime-monitor.go

```diff
@@ -256,26 +256,109 @@ func (monitor *UpTimeMonitorService) processProviderConfig(m models.Monitor, cre
 			body += "&type=2"
 
 			if providerConfig != nil && len(providerConfig.KeywordExists) != 0 {
-
 				if strings.Contains(strings.ToLower(providerConfig.KeywordExists), "yes") {
 					body += "&keyword_type=1"
 				} else if strings.Contains(strings.ToLower(providerConfig.KeywordExists), "no") {
 					body += "&keyword_type=2"
 				}
-
 			} else {
 				body += "&keyword_type=1" // By default 1 (check if keyword exists)
 			}
 
+			// Keyword Value (Required for keyword monitoring)
 			if providerConfig != nil && len(providerConfig.KeywordValue) != 0 {
-				body += "&keyword_value=" + providerConfig.KeywordValue
+				body += "&keyword_value=" + url.QueryEscape(providerConfig.KeywordValue)
 			} else {
 				log.Error(nil, "Monitor is of type Keyword but the `keyword-value` is missing")
 			}
 		}
 	} else {
 		body += "&type=1" // By default monitor is of type HTTP
 	}
+
+	// SubType (Optional for certain types)
+	if providerConfig != nil && len(providerConfig.SubType) != 0 {
+		body += "&sub_type=" + url.QueryEscape(providerConfig.SubType)
+	}
+
+	// Port (Optional for certain types)
+	if providerConfig != nil && providerConfig.Port > 0 {
+		body += "&port=" + strconv.Itoa(providerConfig.Port)
+	}
+
+	// Interval (Optional, in seconds)
+	if providerConfig != nil && providerConfig.Interval > 0 {
+		body += "&interval=" + strconv.Itoa(providerConfig.Interval)
+	} else {
+		body += "&interval=" + strconv.Itoa(DefaultInterval)
+	}
```

Comment:
To be removed (duplicates of lines 237-242), generates an error: "Error: "interval" must be a string"